graphql_compiler/tests/test_compiler.py

```diff
@@ -2842,6 +2842,88 @@ def test_not_contains_op_filter_with_optional_tag(self):
         check_test_data(
             self, test_data, expected_match, expected_gremlin, expected_sql, expected_cypher)
 
+    def test_starts_with_op_filter(self):
+        test_data = test_input_data.starts_with_op_filter()
+
+        expected_match = '''
+            SELECT
+                Animal___1.name AS `animal_name`
+            FROM (
+                MATCH {{
+                    class: Animal,
+                    where: ((name LIKE ({wanted} + '%'))),
+                    as: Animal___1
+                }}
+                RETURN $matches
+            )
+        '''
+        expected_gremlin = '''
+            g.V('@class', 'Animal')
+            .filter{it, m -> (it.name ==~ ('/^' + ($wanted + '.*/')))}
```

Comment:
is this `==~` operation the best way of doing starts and ends with? I can't find much documentation online about it... Just wondering what other possibilities you looked at.

Reply:
Predrag noted that Gremlin has a native startsWith/endsWith function  https://blogs.oracle.com/fadevrel/useful-groovy-string-functions